### PR TITLE
fix error wording for different device/platform

### PR DESCRIPTION
**Pull request by hzheng-ledger**
Jira ticket
https://ledgerhq.atlassian.net/browse/LL-5221

Android Nano S

Screenshot 2021-04-16 at 15 14 08

Android Nano X

Screenshot 2021-04-16 at 15 15 08

IOS Nano S (For Nano X, it is similar)

Screenshot 2021-04-16 at 15 20 09

### CODE REVIEW

File: src/components/TranslatedError.js [hunk 1]
```diff
@@ -18,12 +19,26 @@ export default function TranslatedError({ error, field = "title" }: Props) {
     return null;
   }
   // $FlowFixMe
-  const arg: Object = { message: error.message, ...error };
+  const arg: Object = { message: error.message, returnObjects: true, ...error };
```
Comment (henri-ly) [inline src/components/TranslatedError.js:22]: I'm not sure what does `returnObjects` ?

Comment (hzheng-ledger) [reply src/components/TranslatedError.js:22]: by default , returnObjects= false.
the transition accepts only string. This was the old behaviour. 
Now, I have android/ios, nano s/x for different translation, so the translation is an object (an object contains keys like "ios", "android"....) You can refer to my "common.json " to understand how it works.
Thanks

File: src/locales/en/common.json [hunk 2]
```diff
@@ -105,11 +105,29 @@
     },
     "ConnectAppTimeout": {
       "title": "Sorry, no device found",
-      "description": "Please make sure your {{productName}} is unlocked with Bluetooth enabled."
+      "description": {
+        "ios": {
+          "nanoS": "Please make sure your {{productName}} is unlocked and has Bluetooth enabled.",
```
Comment (LFBarreto) [inline src/locales/en/common.json:110]: this case seems odd :sweat_smile: nanoS can't be connected via bluetooth. we can keep it

File: src/components/TranslatedError.js [hunk 3]
```diff
@@ -18,12 +19,26 @@ export default function TranslatedError({ error, field = "title" }: Props) {
...
+    let translation = t(`errors.${error.name}.${field}`, arg);
     if (translation !== `errors.${error.name}.${field}`) {
-      // It is translated
-      return translation;
+      if (typeof translation === "object" && "productName" in arg) {
+        // it has specific translation for different device and platform
+        const platform = Platform.OS;
+        const device = arg.productName.includes("Nano S") ? "nanoS" : "nanoX";
+        translation = t(
+          `errors.${error.name}.${field}.${platform}.${device}`,
```
Comment (LFBarreto) [inline src/components/TranslatedError.js:31]: maybe put these string id in a const since they are reused to compare afterwards

**Review by LFBarreto — CHANGES_REQUESTED** [review]
I think we could have simplified this by being more generic in the message rather than hacking the specifics.
maybe "Please make sure your {{productName}} is unlocked and connected to your phone." :shrug: 
@dasilvarosa ?

File: src/locales/en/common.json (hunk 2 above)
Comment (juan-cortes) [reply src/locales/en/common.json:110]: there'd never be a case though, why keep the wording ? 🤔

**Review by juan-cortes — APPROVED** [review]
I'm not sure about the approach of the wording, nanoS will not be connected to iOS (unless through a proxy) and the wording about having Bluetooth doesn't make sense in that case. Other than that, if QA is happy, I'm ok.

**Comment by achabrolin-ledger** [comment]
FYI: `yarn flow` fails with 912 errors, unrelated with modified code. Don't know how nor why it fails

**Pull request merged**